### PR TITLE
Feature: Analysis Task - Scan APK file

### PR DESCRIPTION
**Pull request by zhou9584**
## Pull Request Checklist

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Code compiles correctly with all tests are passed.
- [x] I've read the [contributing guide](https://github.com/microsoft/HydraLab/blob/main/CONTRIBUTING.md#making-changes-to-the-code) and followed the recommended practices.
- [ ] [Wikis](https://github.com/microsoft/HydraLab/wiki) or [README](https://github.com/microsoft/HydraLab/blob/main/README.md) have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?
*If this introduces a breaking change for Hydra Lab users, please describe the impact and migration path.*

- [x] Yes
- [ ] No

## How you tested it
*Please make sure the change is tested, you can test it by adding UTs, do local test and share the screenshots, etc.*

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Technical design
- [ ] Build related changes
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Code style update (formatting, renaming) or Documentation content changes
- [ ] Other (please describe): 

### Feature UI screenshots or Technical design diagrams
*If this is a relatively large or complex change, kick it off by drawing the tech design with PlantUML and explaining why you chose the solution you did and what alternatives you considered, etc...*
image
image
image

### CODE REVIEW

**Comment by olivershen-wow** [comment]
Change frontend RUNNER view to reflect abstract assemble, including the title and table strings displayed by clicking RUN button.

**Review by TedaLIEz — APPROVED** [review]

File: agent/src/main/java/com/microsoft/hydralab/agent/runner/appium/AppiumRunner.java [hunk 1]
```diff
@@ -82,7 +83,7 @@ protected File runAndGetGif(File appiumJarFile, String appiumCommand, TestRunDev
                                 TestRun testRun, File deviceTestResultFolder, Logger reportLogger) {
         //set appium test property
         reportLogger.info("Start set appium test property");
-        Map<String, String> instrumentationArgs = testTask.getInstrumentationArgs();
+        Map<String, String> instrumentationArgs = testTask.getTaskRunArgs();
```
Comment (olivershen-wow) [inline agent/src/main/java/com/microsoft/hydralab/agent/runner/appium/AppiumRunner.java:86]: Update variable name accordingly

File: common/src/main/java/com/microsoft/hydralab/common/entity/center/AgentDeviceGroup.java [hunk 2]
```diff
@@ -40,11 +43,25 @@ public void initWithAgentUser(AgentUser agentUser) {
...
     }
 
     public interface Status {
         String HEALTHY = "HEALTHY";
         String ERROR = "ERROR";
         String UPDATING = "UPDATING";
     }
+
+    public void runAnalysisTask(String runnerType) {
+        availableAnalysisTaskCount.put(runnerType, availableAnalysisTaskCount.getOrDefault(runnerType, 3) - 1);
```
Comment (olivershen-wow) [inline common/src/main/java/com/microsoft/hydralab/common/entity/center/AgentDeviceGroup.java:61]: Why 3 here? Maybe move out to a variable for declaration.

File: common/src/main/java/com/microsoft/hydralab/common/entity/center/AgentDeviceGroup.java [hunk 3]
```diff
@@ -40,11 +43,25 @@ public void initWithAgentUser(AgentUser agentUser) {
...
+    }
+
+    public void finishAnalysisTask(String runnerType) {
+        availableAnalysisTaskCount.put(runnerType, availableAnalysisTaskCount.getOrDefault(runnerType, 0) + 1);
```
Comment (olivershen-wow) [inline common/src/main/java/com/microsoft/hydralab/common/entity/center/AgentDeviceGroup.java:65]: Same as above

File: center/src/main/java/com/microsoft/hydralab/center/controller/TestTaskController.java [hunk 4]
```diff
@@ -107,6 +109,9 @@ public Result<Object> runTestTask(@CurrentSecurityContext SysUser requestor,
             return Result.ok(result);
         } catch (HydraLabRuntimeException e) {
             return Result.error(e.getCode(), e);
+        } catch (IllegalArgumentException e) {
```
Comment (olivershen-wow) [inline center/src/main/java/com/microsoft/hydralab/center/controller/TestTaskController.java:112]: What specific scenario is under this catch clause?

File: agent/src/test/java/com/microsoft/hydralab/agent/service/TestTaskEngineServiceTest.java [hunk 5]
```diff
@@ -27,11 +25,12 @@ public class TestTaskEngineServiceTest extends BaseTest {
 
     @Resource
     TestTaskEngineService testTaskEngineService;
-    @MockBean
-    EspressoRunner espressoRunner;
     @Resource
     ApplicationContext applicationContext;
 
+    @Resource
+    TestRunnerManager testRunnerManager;
```
Comment (olivershen-wow) [inline agent/src/test/java/com/microsoft/hydralab/agent/service/TestTaskEngineServiceTest.java:32]: Seems unused here, remove it?

File: center/src/main/java/com/microsoft/hydralab/center/controller/TestTaskController.java (hunk 4 above)
Comment (zhou9584) [reply center/src/main/java/com/microsoft/hydralab/center/controller/TestTaskController.java:112]: Will catch the exception throwed by Asserts

**Review by olivershen-wow — DISMISSED** [review]

**Review by olivershen-wow — APPROVED** [review]

**Review by TedaLIEz — APPROVED** [review]

**Pull request merged**